lib/runtime/wasmer/imports.go

```diff
@@ -185,45 +188,191 @@ func ext_sandbox_memory_teardown_version_1(context unsafe.Pointer, a C.int32_t)
 }
 
 //export ext_crypto_ed25519_generate_version_1
-func ext_crypto_ed25519_generate_version_1(context unsafe.Pointer, a C.int32_t, z C.int64_t) C.int32_t {
+func ext_crypto_ed25519_generate_version_1(context unsafe.Pointer, keyTypeID C.int32_t, seedSpan C.int64_t) C.int32_t {
 	logger.Trace("[ext_crypto_ed25519_generate_version_1] executing...")
-	logger.Warn("[ext_crypto_ed25519_generate_version_1] unimplemented")
-	return 0
+
+	instanceContext := wasm.IntoInstanceContext(context)
+	runtimeCtx := instanceContext.Data().(*runtime.Context)
+
+	// TODO: key types not yet implemented
+	// id := memory[idData:idData+4]
+
+	seedBytes := asMemorySlice(instanceContext, seedSpan)
+	buf := &bytes.Buffer{}
+	buf.Write(seedBytes)
+
+	seed, err := optional.NewBytes(false, nil).Decode(buf)
+	if err != nil {
+		logger.Warn("[ext_crypto_ed25519_generate_version_1] cannot generate key", "error", err)
+		return 0
+	}
+
+	var kp crypto.Keypair
+
+	if seed.Exists() {
+		kp, err = ed25519.NewKeypairFromSeed(seedBytes)
+	} else {
```

Comment:
this is a bit of a misnomer- according to the spec, the seed is a BIP39 mnemonic seed that's used to derive the key, whereas the function `NewKeypairFromSeed` generates a keypair using the seed as a random seed. you can leave it like this for now, since the BIP39 functionality hasn't been added to gossamer yet (but it's implemented here https://github.com/ChainSafe/go-schnorrkel/blob/master/bip39.go) I will open an issue for this as well

Reply:
maybe just add a comment here saying this needs to be updated to use BIP39 derivation